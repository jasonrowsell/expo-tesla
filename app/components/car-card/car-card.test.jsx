import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { Alert } from 'react-native';

import CarCard from './car-card';

describe('<CarCard />', () => {
  const renderComponent = () => render(
    <CarCard title="title" tagline="tagline" taglineCTA="taglineCTA" source="img" />,
  );

  test('has 3 children', () => {
    const tree = renderer.create(<CarCard title="title" tagline="tagline" source="img" />).toJSON();
    expect(tree.children.length).toBe(3);
  });

  test('renders correctly', () => {
    const tree = renderComponent();
    expect(tree).toMatchSnapshot();
  });

  test('has the title', () => {
    const { getByText } = renderComponent();
    getByText('title');
  });

  test('has the tagline followed by the CTA', () => {
    const { getByText } = renderComponent();
    getByText('tagline taglineCTA');
  });

  test('custom order alert is displayed when custom order button is pressed', () => {
    jest.spyOn(Alert, 'alert');
    const { getByText } = renderComponent();

    const button = getByText('Custom Order');
    fireEvent.press(button);

    expect(Alert.alert).toHaveBeenCalledWith('Custom order was pressed');
  });

  test('existing inventory alert is displayed when existing inventory button is pressed', () => {
    jest.spyOn(Alert, 'alert');
    const { getByText } = renderComponent();

    const button = getByText('Existing Inventory');
    fireEvent.press(button);

    expect(Alert.alert).toHaveBeenCalledWith('Existing inventory was pressed');
  });
});
