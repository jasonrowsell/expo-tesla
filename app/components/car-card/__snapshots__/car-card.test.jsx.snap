// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarCard /> renders correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "position": "absolute",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={null}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": "30%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "500",
        }
      }
    >
      title
    </Text>
    <Text
      style={
        Object {
          "color": "#5c5e62",
          "fontSize": 16,
        }
      }
    >
      tagline
       
      <Text
        style={
          Object {
            "textDecorationLine": "underline",
          }
        }
      >
        taglineCTA
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "bottom": 50,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#171A20CC",
            },
          ]
        }
        testID="button"
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 12,
                "fontWeight": "500",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Custom Order
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#FFFFFFA6",
            },
          ]
        }
        testID="button"
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 12,
                "fontWeight": "500",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#171A20",
              },
            ]
          }
        >
          Existing Inventory
        </Text>
      </View>
    </View>
  </View>
</View>
`;
